### DISCUSSION TITLE
Release 0.6.11 (1.21.30.22)

### DISCUSSION THREAD

**Discussion by m-brisch** [Mojang/minecraft-editor#188] [Announcements]
_[Read the previous release notes here](https://github.com/Mojang/minecraft-editor/discussions/187)_  

---    

## Summary  

Hello and happy preview day 🥳! We are now on 0.6.11 (1.21.30.22), as of Wednesday, August 7th, 2024. 

As a reminder of Editor API support:   

- Editor Extensions [overview article](https://learn.microsoft.com/en-us/minecraft/creator/documents/editorextensionsintroduction) and [API reference documentation](https://aka.ms/EditorAPI) on learn.microsoft.com  
- [Starter Kit](https://github.com/Mojang/minecraft-editor-extension-starter-kit): Contains the compiler tools and various libraries required to get started making Editor Extensions  
- [Samples](https://github.com/Mojang/minecraft-editor-extension-samples) created by our team to jump-start your own Extension creations  

Additionally, if you're new to the Editor GitHub this week, we encourage you to join the Bedrock Add-Ons Discord [#Editor channel](https://discord.gg/TZnQKFwGpt)!  

---

## Deferred Lighting Settings update
We added a new `Water` dropdown to the Deferred Lighting Settings panel!

https://github.com/user-attachments/assets/62718c3a-3d98-47c5-b998-654d0d1d0242

You can learn more about the settings on the [Deferred Lighting Learning page](https://learn.microsoft.com/en-us/minecraft/creator/documents/deferredlighting/watercustomization?view=minecraft-bedrock-stable)

---

## Export Pane 
We added a new panel to our export flow so you're able to customize the world settings before exporting to file
![image](https://github.com/user-attachments/assets/7f906298-d6e6-4d0b-a336-330f3066d7c2)

---

## Updated Brush Tool
We made several improvements to the brush pipeline which should make things much more performant. There's a new `Single Block` brush shape for detailed work. We also added `Masking Mode` which allows you to enumerate types of blocks you want the brush to effect or ignore.

Here's an example of using the replace mask to only apply the brush to the `acacia_leaves`:

https://github.com/user-attachments/assets/86ba9ae9-9964-4ce8-a8c4-749ed99ed72e

---

## Ruler Tool
We added a new ruler tool to help plan out builds and show what the new client widget system can do.

https://github.com/user-attachments/assets/4be57a82-b710-4b1c-bd82-6e50bfe1b4b2

---

## Full Change Log   
- Added `Water` dropdown inside the `Deferred Lighting Settings` panel.
- Added `Export` pane inside the file menu to allow customizing world settings before exporting to file.
- Updated Brush Tool with additional features and improved performance.
  - Added `Single Block` brush shape to allow more fine grain editing.
  - Added `Masking Mode` to customize how brushes interact with the world.
    - `Replace` will create a list of blocks that will be replaced when brushing. Blocks not in this list will not be impacted.
    - `Mask` will create a list of blocks that will be masked/ignored when brushing. Only blocks not in the list will be impacted.
- Added `IRootPropertyPane` and `ISubPanePropertyItem` interfaces to improve property pane API usability
  - Renamed `createPropertyPane` and `removePropertyPane` functions for `IPropertyPane` to `createSubPane` and `removeSubPane`.
  - Added action button support for root property pane header.
  - Added visual customization support for sub panes.
- Added Ruler tool for measuring distances with new client widget framework.
- Added constraint on the weight value in a Block palette item. Valid values are in the range [1,100].
- Added missing Widget binding export 
---   

Thank you for being here! You're the best! ❤️

**Comment by EthanMcknney** [comment]
Can we add an update to the villages and villagers like MCA?
